src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,594 @@
+                            containing grant information
+        grantInd:           indices of grant requests
+        exclusionZone:      a polygon object or a list of locations.
+                            If it is a list of locations, each one being a tuple
+                            with fields 'latitude' and 'longitude'.
+        dpaType:            an enum member of class DpaType
+        hIncAnt:            reference incumbent antenna height (m)
+        numIter:            number of Monte Carlo iterations
+        threshold:          protection threshold (dBm/10 MHz)
+        beamwidth:          protection antenna beamwidth (degree)
+
+    Returns:                the list of indices of grants on the move list for the
+                            given protection constraint
+    """
+
+    # Assign values to the protection constraint
+    constraint = ProtectionConstraint(latitude=protectionPoint.latitude,
+                                      longitude=protectionPoint.longitude,
+                                      lowFrequency=lowFreq,
+                                      highFrequency=highFreq)
+
+    # Identify Nc CBSD grants in the neighborhood of the protection constraint
+    Nc_grants = findGrantsInsideNeighborhood(registrationReq, grantReq, grantInd,
+                                             constraint, exclusionZone, dpaType)
+
+    if len(Nc_grants):  # Found CBSDs in the neighborhood
+
+        # Form the matrix of interference contributions
+        I, sorted_grant_indices, bearings = formInterferenceMatrix(Nc_grants, constraint,
+                                                                hIncAnt, numIter, dpaType)
+
+        # Find the index (nc) of the grant in the ordered list of grants such that
+        # the protection percentile of the interference from the first nc grants is below
+        # the threshold for all azimuths of the receiver antenna.
+        nc = find_nc(I, bearings, threshold, beamwidth)
+
+        # Determine the associated move list (Mc)
+        Mc = sorted_grant_indices[nc:]
+
+        return Mc
+
+    else:
+        
+        return []
+
+
+def findMoveList(protection_specs, protection_points, registration_requests,
+                 grant_requests, num_iter, num_processes, exclusion_zone=None):
+
+    """
+    Main routine to find CBSD indices on the move list.
+
+    Inputs:
+        protection_specs:   a list of protection specifications, a named tuple with
+                            fields of 'lowFreq' (in Hz), 'highFreq' (in Hz),
+                            'antHeight' (in meter), 'beamwidth' (in degree),
+                            'threshold' (in dBm/10MHz)
+        protection_points:  a list of protection points, each one being a tuple with
+                            named fields of 'latitude' and 'longitude'
+        registration_requests: a list of CBSD registration requests, each one being
```

Comment:
With the way that the test cases are written in the test spec, we will always be using information in the FAD (Full Activity Dump) format as input to the DPA move list reference model. A useful interface to this reference model would be something like:

Inputs:

- protection_specs
- protection_points
- SAS UUT FAD information
- List of FADs from SAS Test Harnesses

Output:

- Move list for SAS UUT, format:  [ (cbsd_id1, [grant_id1, grant_id4]), (cbsd_id3, [grant_id2]) ]
- List of move lists for SAS Test Harnesses (i.e. a list of lists, where each list is in the format above)

One option is to write a wrapper around this existing function, perhaps in a followup PR.

Reply:
Agree with the suggestion to write a wrapper.

Reply:
Great! Is there any chance I can assume that NIST would do this? :)

Reply:
Sure, we can write a wrapper.  Can you point us to agreed upon formats for the inputs?

Just to clarify, there is only one output move list, not a move list for the SAS UUT and a separate move list for each SAS Test Harness.  Are you asking for this single output list of grants to be segmented by their managing SAS?

Reply:
Thank you! I will send you the relevant information via email and I'll be happy to answer any clarifying questions (or redirect you to the right party).

I think that in the end we'll want to segment the move lists based on originating SAS so that we can use them directly in e.g. MCP.1. At the very least we'll need to distinguish between "SAS UUT ML" and "union of MLs for all SAS Test Harnesses", and I think we'll find it most convenient to split them fully.